json_rpc/clients/httpclient.nim

```diff
@@ -34,12 +37,10 @@ proc sendRequest(transp: StreamTransport,
   if len(data) > 0:
     request.add(data)
   try:
-    let res = await transp.write(cast[seq[char]](request))
-    if res != len(request):
-      result = false
-    result = true
-  except:
-    result = false
+    let res = await transp.write(request.toBytes())
+    return res == len(request):
+  except CatchableError:
```

Comment:
If you want to use `CatchableError` handler only you, you will catch `CancelledError` too.
Add `except CancelledError: raise exc` too.

Reply:
fixed, though the exception issues in the code run much deeper than this